pkg/controller/node/node_controller.go

```diff
@@ -477,19 +498,15 @@ func (r *ReconcileNode) Reconcile(request reconcile.Request) (reconcile.Result,
 		return reconcile.Result{}, err
 	}
 
-	nodeAddresses := instance.Status.Addresses
-	var nodeAddress string
-	// TODO: An example for the nodeAddresses is:
-	// [{ExternalIP 192.168.10.38} {InternalIP 192.168.10.38} {Hostname compute-0}]"}
-	// Need to confirm if it's possible to have multiple ExternalIPs and handle that case
-	reqLogger.Info("Got the node address info", "nodeAddresses", nodeAddresses)
-	for _, address := range nodeAddresses {
-		if address.Type == "ExternalIP" {
-			nodeAddress = address.Address
-			break
+	nodeAddressesWithType := instance.Status.Addresses
+	var nodeAddresses []string
```

Comment:
Right now I'm thinking as we use node IP address to get node's LSP, it has a limitation that k8s cluster is using nsx-attached vNIC. In this case I guess there is only one IP for the node, either in ExternalIP or InternalIP? But using `nodeAddresses` as a list here should be no harm.

Reply:
Yes, currently we only have one IP. I use the list because of the possible expansion in the future.